source/common/logger/ur_logger.h

```diff
@@ -0,0 +1,52 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: MIT
+
+#include "ur_level.h"
+#include "ur_sinks.h"
+
+namespace logger {
+
+class Logger {
+  public:
+    Logger(sink::Sink *sink) : sink(sink) {
```

Comment:
I think it would be better to pass `sink` as `std::unique_ptr<Sink>` - ther logger takes ownership of the sink, right?

Reply:
Done